backend/src/api/alias.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Context;
+
+use axum::{routing::get, Router};
+use axum::{Extension, Json};
+use serde::{Deserialize, Serialize};
+
+use crate::api::error::Error;
+use crate::{AppState, DbInteractable, FileDb, SqliteDatabase};
+
+use super::auth::Authorize;
+use std::sync::Arc;
+
+use serde_rusqlite::from_row;
+
+use rusqlite::params;
+
+pub fn api_route() -> Router {
+    Router::new().route("/", get(get_aliases::<FileDb>))
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Alias {
+    name: String,
+    content: String,
+}
+
+async fn get_aliases<D: SqliteDatabase>(
+    Authorize(_username): Authorize,
+    Extension(state): Extension<Arc<AppState<D>>>,
+) -> Result<Json<Vec<Alias>>, Error> {
+    let conn = state.pool.get().await.context("Failed to get connection")?;
+
+    conn.interact(move |conn| {
+        let mut query = conn
+            .prepare("SELECT * FROM ALIASES")
```

Comment:
I'm surprised to see that table names are case insensitive but I would prefer if only SQL commands were uppercase and table names were lowercase like in the database.

Reply:
I'd argue that for this specific request I'd want to have all the available information for aliases.

Reply:
That's not how it works though, you have to make code changes to the `Alias` struct anyway if you want to serve more data: https://github.com/Mavulp/hivefriends/blob/0aa883a0f4569f37c5410107e0c395f9d09fc3b1/backend/src/api/alias.rs#L22-L25